src/test/regress/output/multi_alter_table_statements.source

```diff
@@ -552,6 +552,95 @@ SELECT master_create_worker_shards('single_shard_items', 1, 2);
  
 (1 row)
 
+-- Verify that ALTER TABLE .. REPLICATION IDENTITY [USING INDEX]* .. works
+ALTER TABLE single_shard_items ALTER COLUMN id SET NOT NULL;
```

Comment:
Why do we execute `ALTER TABLE single_shard_items ALTER COLUMN id SET NOT NULL;` here?

Reply:
If the question is "Is it necessary the column to be not null" the answer is: Indexes cannot be used as replica identity if the column is nullable.
But if the question is "isn't it better we make the column not null when we are creating the table?", the answer is yes. You are right, I'm moving the definition to there.